test/helpers/types.go

```diff
@@ -99,3 +102,102 @@ func (l logs) NonEmpty() bool {
 	}
 	return len(l) > 0
 }
+
+func isNil(i interface{}) bool {
+	if i == nil {
+		return true
+	}
+	switch reflect.TypeOf(i).Kind() {
+	case reflect.Ptr, reflect.Map, reflect.Array, reflect.Chan, reflect.Slice:
+		return reflect.ValueOf(i).IsNil()
+	}
+	return false
+}
+
+func DeepFields(iface interface{}, namePrefix string) ([]reflect.Value, []string) {
```

Comment:
We should add tests to exercise this code

Reply:
Done